crates/rattler_installs_packages/src/python_env/system_python.rs

```diff
@@ -8,16 +9,42 @@ use thiserror::Error;
 pub enum FindPythonError {
     #[error("could not find python executable")]
     NotFound,
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
 }
 
 /// Try to find the python executable in the current environment.
+/// Using sys.executable aproach will return original interpretator path
+/// and not the shim in case of using which
 pub fn system_python_executable() -> Result<PathBuf, FindPythonError> {
     // When installed with homebrew on macOS, the python3 executable is called `python3` instead
     // Also on some ubuntu installs this is the case
     // For windows it should just be python
-    which::which("python3")
-        .or_else(|_| which::which("python"))
-        .map_err(|_| FindPythonError::NotFound)
+
+    let output = match std::process::Command::new("python3")
```

Comment:
Given the move away from a few stat calls to a subprocess, consider caching the results of `system_python_executable` across the entire process since it's unlikely to change and you can avoid this overhead on every environment creation + bytecode compile call etc.

Reply:
Thanks @pradyunsg will make an issue out of this!